evaltools/utils/JSON.py

```diff
@@ -1,20 +1,56 @@
 
 import json
-from typing import List, Union
+from pydantic import BaseModel, ValidationError, validator
+from typing import List, Union, Any
 
-class JSON(object):
+class JSON(BaseModel):
     """
-    A class which represents JSON data as a Python object rather than a dict. If
-    the JSON object is a list, returns a list of JSON-ified objects.
+    Plan specification models. To better work with multiple plans at once, this
+    plan specification allows users to specify information which should remain
+    consistent across all operations; for example, the `column` field should be
+    the name of the column in which the corresponding plan's assignment is stored
+    across all data products.
     """
-    def __init__(self, data):
+
+    column: str
+    """
+    Column on all data products which contains district assignment information
+    for this districting plan.
+    """
+
+    locator: str
+    """
+    File/directory name for all resources which contain information about this
+    districting plan.
+    """
+
+    title: Any = None
+    """
+    Official title of the plan.
+    """
+
+    @validator("column")
+    def _validate_column(cls, column):
         """
-        Args:
-            data (string): Data to jsonify.
+        Validates the specified column name. For most of our purposes, we cannot
+        include spaces, hyphens, percent signs, and other commonly-used special
+        characters. Furthermore, because these plans are often saved in shapefiles,
+        which have a 10-character column name limit, we force column names to be
+        10 or fewer characters.
         """
-        for k, v in data.items(): self.__dict__[k] = v
+        # Check for column length.
+        if len(column) > 10: raise ValueError(f"Column name {column} exceeds 10-character limit.")
+
+        # Check for illegal characters.
+        illegal = {"/", "%", "-", "–", "—", " "}
+        
+        for c in illegal:
+            if c in column: raise ValueError(f"Character {c} cannot be in column name.")
 
-def objectify(location) -> Union[List[JSON],JSON]:
+        return column
+
+
+def jsonify(location) -> list:
```

Comment:
The types could be more specific here (which will help with IDE auto-completion):

```
from typing import List

def jsonify(location: str) -> List[JSON]
```